R/clean_data_helpers.R

```diff
@@ -1,76 +1,147 @@
-#' Calculate the percentage of missing and other data type values in a vector
-#' containing different data types such as numeric, Date, character, and
-#' logical.
+#' Scan through all character columns of a data frame to determine the
+#' proportion of `missing`, `numeric`, `Date`, `character`, `logical`, values.
 #'
-#' @param x A vector of ones or a combination of various data types.
+#' @param data A data frame or linelist
+#'
+#' @returns A data frame with the same number of rows as the number of character
+#'    columns of the input data, and six (06) columns representing the field
+#'    names, the percentage of missing, numeric, date, character, and logical,
+#'    values in each column.
+#'
+#' @export
+#'
+#' @examples
+#' # scan through a data frame of characters
+#' scan_result <- scan_data(
+#'   data = readRDS(system.file("extdata", "messy_data.RDS",
+#'                              package = "cleanepi"))
+#' )
+#'
+#' # scan through a data frame with two character columns
+#' scan_result <- scan_data(
+#'   data = readRDS(system.file("extdata", "test_linelist.RDS",
+#'                              package = "cleanepi"))
+#'   )
+#'
+#' # scan through a data frame with no character columns
+#' data(iris)
+#' iris[["fct"]] <- as.factor(sample(c("gray", "orange"), nrow(iris),
+#'                            replace = TRUE))
+#' iris[["lgl"]]  <- sample(c(TRUE, FALSE), nrow(iris), replace = TRUE)
+#' iris[["date"]] <- as.Date(seq.Date(from = as.Date("2024-01-01"),
+#'                                    to = as.Date("2024-08-30"),
+#'                                    length.out = nrow(iris)))
+#' iris[["posit_ct"]] <- as.POSIXct(iris[["date"]])
+#' scan_result        <- scan_data(data = iris)
+#'
+scan_data <- function(data) {
+  # scan through all columns of the data and the identify character columns
+  types <- vapply(data, typeof, character(1L))
+  target_columns <- types[types == "character"]
+
+  # send an message if there is no character column found within the input data
+  if (length(target_columns) == 0L) {
+    message("No character column found in the provided data.")
+    return(invisible(NA))
```

Comment:
The message would have been enough for this. I am returning `NA` here so that in order to have a value that will be used to know whether the scanning result will be displayed or not in the `print_report()` function.
`Invisibly` because I am masking the NA from the user's console as the message is informative enough.